### PR TITLE
refactor(tokens): move less variables to css props

### LINKED ISSUE #1992: Move less variable to CSS props
Major version LESS/CSS Tokens/Prop Changes

Move more `LESS` vars to `CSS` props. Which ones? TBD.

### PR DESCRIPTION
**Pull request by saiponnada**
Fixes #1992 

- [x] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
Refactored following less variables to css variables
-  @font-size-*
-  @font-size-{variant}
-  @font-weight-*
-  @spacing-*

Left the rest(@_screen-size-*, @_page-grid-*, @marketsans-*) as the first two are constants and we don't want to allow overrides and the later due to restriction in css vars.

## Notes
-  For `font-size`, we have flexibility with defining sizes(12, 14, 16...) and variants(like small, med, large..). It makes perfect sense using it in less. Assuming components use only variants, I personally feel there is no advantage of having that flexibility in css vars because its just a 1:1 mapping in tokens file. Besides that we will be sending all these tokens (more lines of code). While review, please state your preference to leave both as is or to simplify it. If we simplify it we need to identify the right variant tokens for combobox and select which are currently using `--font-size-18` 

## Checklist

- [x] I verify the build is in a non-broken state
- [x] I verify all changes are within scope of the linked issue

- [x] I regenerated all CSS files under dist folder
- [x] I tested the UI in all supported browsers
- [x] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [x] I tested the UI in dark mode and RTL mode
- [ ] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

File: src/less/combobox/combobox.less [hunk 1]
```diff
@@ -181,7 +181,7 @@ span.combobox {
 }
 
 .combobox--large .combobox__control > input {
-    font-size: @font-size-18;
+    font-size: var(--font-size-18);
```
Comment (saiponnada) [inline src/less/combobox/combobox.less:184]: Directly using `--font-size-18` instead of a variant here

File: src/less/select/select.less [hunk 2]
```diff
@@ -35,7 +35,7 @@ span.select {
 }
 
 .select--large select {
-    font-size: @font-size-18;
+    font-size: var(--font-size-18);
```
Comment (saiponnada) [inline src/less/select/select.less:38]: Directly using `--font-size-18` instead of a variant here

File: src/tokens/evo-core.css [hunk 3]
```diff
@@ -109,4 +109,15 @@
     --font-size-giant-4: var(--font-size-64);
     --font-weight-regular: 500;
     --font-weight-bold: 700;
+    --spacing-100: 8px;
+    --spacing-25: calc(var(--spacing-100) * 0.25);
```
Comment (saiponnada) [inline src/tokens/evo-core.css:113]: May be computing the value `calc(var(--spacing-100) * 0.25)` to `2px` might be better here. any thoughts?

File: src/less/variables/variables.less [hunk 4]
```diff
@@ -19,51 +19,14 @@
...
+@_page-grid-row-gaps-large: var(--spacing-200);
 
 @marketsans-url: "https://ir.ebaystatic.com/cr/v/c1/market-sans/v1.0/";
 @marketsans-fontname: "Market Sans";
 @marketsans-filename-regular: "MarketSans-Regular-WebS";
 @marketsans-filename-bold: "MarketSans-SemiBold-WebS";
 
 @font-family-default: @marketsans-fontname, Arial, sans-serif;
-
-@font-weight-regular: 500;
```
Comment (ianmcburnie) [inline src/less/variables/variables.less:34]: We can't remove these until next major version.

**Comment by ianmcburnie** [comment]
One thing worthy of discussion is whether we want to continue with the token shorthand mixin that we were using for colors. or move away from that, e.g.

```
.color-token(filter-menu-item-foreground-color, color-foreground-primary);
```

File: src/less/variables/variables.less (hunk 4 above)
Comment (saiponnada) [reply src/less/variables/variables.less:34]: Done. rolled back.

File: src/tokens/evo-core.css (hunk 3 above)
Comment (ianmcburnie) [reply src/tokens/evo-core.css:113]: Yes, `2px` here would be fine. Don't think we need to be too clever.

**Comment by saiponnada** [comment]
> One thing worthy of discussion is whether we want to continue with the token shorthand mixin that we were using for colors. or move away from that

Team has discussed internally and decided to continue with the token mixin the way we are using for now. Need to have a separate discussion later on how we would like to define or expose API for component level by aligning with skin's long term vision.

File: src/less/select/select.less (hunk 2 above)
Comment (saiponnada) [reply src/less/select/select.less:38]: Would like to create a new variant for consistency `--font-size-medium-2` or `--font-size-large-0` but there will be a mismatch with DS. Should I go this approach by talking to DS?

Comment (saiponnada) [reply src/less/select/select.less:38]: Created a new issue to fix this later #2175

File: src/less/combobox/combobox.less (hunk 1 above)
Comment (saiponnada) [reply src/less/combobox/combobox.less:184]: Created a new issue to fix this later #2175

**Review by ArtBlue — APPROVED** [review]
Looks good. Assuming we resolve the conflict in the branch and things still look good, this is approved.

**Pull request merged**